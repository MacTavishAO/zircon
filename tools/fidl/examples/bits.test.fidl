// Copyright 2019 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

library fidl.examples.bits;

bits uint8_bits : uint8 {
    kOne = 1;
    kTwo = 2;
    kFour = 4;
};

bits uint16_bits : uint16 {
    kOne = 0x1;
    kTwo = 2;
    kFour = 4;
};

bits uint32_bits : uint32 {
    kOne = 1;
    kTwo = 2;
    kFour = 4;
    kEight = 8;
};

bits uint64_bits : uint64 {
    kOne = 1;
    kTwo = 2;
    kFour = 4;
    kEight = 8;
    kSixteen = 0x10;
    kThirtyTwo = 0x20;
    kBiggestBit = 0x8000000000000000;
};
