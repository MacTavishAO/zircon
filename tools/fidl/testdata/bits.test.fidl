library fidl.test.json;

bits MyBits : uint32 {
    MY_FIRST_BIT = 1;
    MY_OTHER_BIT = 2;
};

bits StrictBits : uint64 {
    SMALLEST = 1;
    BIGGEST = 0x8000000000000000;
};
