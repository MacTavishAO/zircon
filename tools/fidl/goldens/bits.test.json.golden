{
  "version": "0.0.1",
  "name": "fidl.test.json",
  "library_dependencies": [],
  "bits_declarations": [
    {
      "name": "fidl.test.json/MyBits",
      "location": {
        "filename": "bits.test.fidl",
        "line": 3,
        "column": 6,
        "length": 6
      },
      "type": {
        "kind": "primitive",
        "subtype": "uint32"
      },
      "mask": "3",
      "members": [
        {
          "name": "MY_FIRST_BIT",
          "location": {
            "filename": "bits.test.fidl",
            "line": 4,
            "column": 5,
            "length": 12
          },
          "value": {
            "kind": "literal",
            "value": "1",
            "expression": "1",
            "literal": {
              "kind": "numeric",
              "value": "1",
              "expression": "1"
            }
          }
        },
        {
          "name": "MY_OTHER_BIT",
          "location": {
            "filename": "bits.test.fidl",
            "line": 5,
            "column": 5,
            "length": 12
          },
          "value": {
            "kind": "literal",
            "value": "2",
            "expression": "2",
            "literal": {
              "kind": "numeric",
              "value": "2",
              "expression": "2"
            }
          }
        }
      ],
      "strict": true
    },
    {
      "name": "fidl.test.json/StrictBits",
      "location": {
        "filename": "bits.test.fidl",
        "line": 8,
        "column": 6,
        "length": 10
      },
      "type": {
        "kind": "primitive",
        "subtype": "uint64"
      },
      "mask": "9223372036854775809",
      "members": [
        {
          "name": "SMALLEST",
          "location": {
            "filename": "bits.test.fidl",
            "line": 9,
            "column": 5,
            "length": 8
          },
          "value": {
            "kind": "literal",
            "value": "1",
            "expression": "1",
            "literal": {
              "kind": "numeric",
              "value": "1",
              "expression": "1"
            }
          }
        },
        {
          "name": "BIGGEST",
          "location": {
            "filename": "bits.test.fidl",
            "line": 10,
            "column": 5,
            "length": 7
          },
          "value": {
            "kind": "literal",
            "value": "9223372036854775808",
            "expression": "0x8000000000000000",
            "literal": {
              "kind": "numeric",
              "value": "9223372036854775808",
              "expression": "0x8000000000000000"
            }
          }
        }
      ],
      "strict": true
    }
  ],
  "const_declarations": [],
  "enum_declarations": [],
  "experimental_resource_declarations": [],
  "interface_declarations": [],
  "service_declarations": [],
  "struct_declarations": [],
  "table_declarations": [],
  "union_declarations": [],
  "type_alias_declarations": [],
  "declaration_order": [
    "fidl.test.json/StrictBits",
    "fidl.test.json/MyBits"
  ],
  "declarations": {
    "fidl.test.json/MyBits": "bits",
    "fidl.test.json/StrictBits": "bits"
  }
}
